gr-blocks/lib/annotator_raw_impl.h

```diff
@@ -13,16 +13,39 @@
 
 #include <gnuradio/blocks/annotator_raw.h>
 #include <gnuradio/thread/thread.h>
+#include <type_traits>
+#include <set>
+#include <tuple>
 
 namespace gr {
 namespace blocks {
 
 class annotator_raw_impl : public annotator_raw
 {
 private:
-    const size_t d_itemsize;
-    std::vector<tag_t> d_queued_tags;
+    struct tag_comparator {
+        using is_transparent = std::true_type;
+        //!\brief comparator over all fields in a tag, not just the offset
```

Comment:
Is that necessary? An offset compare is much cheaper than the other compares. The offset compare is clearly important. As for the others, if the `d_queued_tags` object stores tags with identical order in any offset, then that seems fine to me. So what happens if we just compare the offset in the comparator and the offset is the same? Seems to me that the `<` would just return false and handle it appropriately.

Reply:
(i.e., if we can't strictly order tags, we can't do things like "all tags with offsets between this and that"; we'd need to create a dummy tag just too look for things, or we again need to linearly search through the container of all tags, which negates the advantages of using a set instead a vector)